.github/workflows/ci-build.yml

```
@@ -72,9 +72,19 @@ jobs:
           name: build-log-java-${{ matrix.java }}-${{ matrix.os }}
           path: build.log
 
+      - name: Change Files
+        run: |
+          echo modification >> CONTRIBUTING.md
+          touch new-file
+          mkdir new-dir
+          touch new-dir/new-file
+          rm README.md
+
       - name: Verify Changed Files
-        uses: tj-actions/verify-changed-files@v20
         id: verify-changed-files
+        run: |
+          changed_files=$((git diff --name-only ${{ github.sha }}; git ls-files --others --exclude-standard) | tr '\n' ' ')
```

```suggestion
          changed_files=$(echo -n "$(git diff --name-only HEAD 2>/dev/null ; git ls-files --others --exclude-standard 2>/dev/null)"|tr '\n' ' ')
```

This seems to work...., see also this build
https://github.com/openhab/openhab-distro/actions/runs/13883197333/job/38844457011?pr=1733

I did deal with the pipefail....

---

Yes that seems to fix it. The echo and subshell, pipe issues can probably be prevented if you use `<<<`  to provide input to `tr` see:

https://unix.stackexchange.com/a/80372

---

I tried to use <<<, but
```
echo X$(tr '\n' ' ' <<< $(git diff --name-only HEAD 2>/dev/null ; git ls-files --others --exclude-standard 2>/dev/null))X
``` returns `X X` (one space between X).

```
echo X$(echo -n "$(git diff --name-only HEAD 2>/dev/null ; git ls-files --others --exclude-standard 2>/dev/null)"|tr '\n' ' ')X
```
returns `XX`.